lib/logwriter.c

```diff
@@ -1150,7 +1150,7 @@ log_writer_write_message(LogWriter *self, LogMessage *msg, LogPathOptions *path_
   else
     {
       msg_debug("Can't send the message rewind backlog",
-                evt_tag_str("message",self->line_buffer->str));
+                evt_tag_printf("message", "%.*s", (gint) self->line_buffer->len, self->line_buffer->str));
```

Comment:
This change is not needed IMO. Internally we will write into a 1024 buffer with vsnprinf anyway. So I do not see danger for overflow here.

Reply:
Good catch. I also modified the original printf in the "true branch".